modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/connectivity/oauth/OAuthConnectionProviderObjectBuilder.java

```diff
@@ -237,7 +256,10 @@ private void withCustomParameters(BiConsumer<ParameterModel, OAuthParameterModel
   }
 
   private String resolveString(CoreEvent event, ValueResolver resolver) throws MuleException {
-    Object value = resolver.resolve(from(event));
+    Object value = resolver.resolve(ValueResolvingContext.builder(event)
+        .dynamic(resolver.isDynamic())
+        .withExpressionManager(expressionManager)
```

Comment:
The only purpose of the expression manager is to be used when the expression is dynamic to create the session, but the builder API doesn't communicate that binding.